### ISSUE TITLE
code optimization

### ISSUE THREAD

**Issue by Uttam1230998897987879** [deepseek-ai/DeepSeek-V3#416]
`import torch.distributions as dist

def sample(logits, temperature=1.0):
    probs = torch. SoftMax(logits / temperature, dim=-1)
    dist = dist.Categorical(probs=probs)
    return dist.sample()
`

**Comment by Some1and2-XC** [comment]
You'll find that GitHub implements this feature that allows you to request merging your code into a repo. This is referred to a pull request. Sometimes maintainers prefer people to use those instead of pasting code snippets into their GitHub issues.
